src/tidy_algo.rs

```diff
@@ -0,0 +1,218 @@
+        self.name == other.name
+    }
+}
+
+pub struct TidyAlgo {
+    rules: Vec<TidyRule>,
+}
+
+impl TidyAlgo {
+    pub const fn new() -> Self {
+        Self { rules: Vec::new() }
+    }
+
+    fn add_rule(&mut self, rule: TidyRule) {
+        self.rules.push(rule);
+    }
+
+    /// Load a rule into the ruleset
+    ///
+    /// Returns either the name of the loaded rule or an Error
+    fn load_rule_from_hashmap(
+        &mut self,
+        table: HashMap<String, Value>,
+    ) -> Result<String, Box<dyn Error>> {
+        let name = get_string_from_table_safe(&table, "name")?;
+        let log = get_string_from_table_safe(&table, "log")?;
+        let scope = get_string_from_table_safe(&table, "scope")?;
+        let apply_type = get_string_from_table_safe(&table, "type")?;
+        let apply = match apply_type.as_str() {
+            "duplicated" => duplicated::aply_duplicated,
+            "misnamed" => misnamed::apply_misnamed,
+            "perished" => perished::apply_perished,
+            fallback => return Err(format!("Could not load rule with type {}", fallback).into()),
+        };
+        debug!(
+            "Adding rule {} of type {} that will be logged as {}",
+            name, apply_type, log
+        );
+        self.add_rule(TidyRule::new(name.clone(), log, scope, table, apply));
+        Ok(name)
+    }
+
+    /// Load rules from a file
+    ///
+    /// Returns the number of rules loaded of an error
+    pub fn load_rules_from_file(
+        &mut self,
+        _my_files: &MyFiles,
+        path: path::PathBuf,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        let rules_config: Result<Config, ConfigError> =
+            Config::builder().add_source(File::from(path)).build();
+
+        let rules = match rules_config {
+            Ok(config) => config.get_array("rules").unwrap(),
+            Err(error) => return Err(format!("Error while loading rules: {}", error).into()),
+        };
+
+        for rule in rules {
+            let table = rule.into_table().unwrap();
+            let _ = self.load_rule_from_hashmap(table);
+        }
+        Ok(self.rules.len())
+    }
+
+    /// Apply the rules to a file
+    pub fn apply_rules(&self, file: &mut FileInfo, my_files: &MyFiles) {
+        for rule in &self.rules {
+            // This unwrap will be handled when we will tackle the permission problem
+            let rule_pathbuf: PathBuf = fix_canonicalize_path(Into::<PathBuf>::into(rule.scope.clone()).canonicalize().unwrap());
+            if file.path.starts_with(&rule_pathbuf) || rule.scope == "all" {
+                let rule_score = (rule.apply)(file, my_files, rule.params.clone());
+                info!("Applied all rules to file {}", file.path.display());
+                file.tidy_score = Some(rule_score);
+            } else {
+                debug!("Skipping {} for rule {} with scope {}", rule.name, file.path.display(), rule_pathbuf.display());
+            }
+        }
+    }
+}
```

Comment:
Currently, we are only applying regex to my_files, correct? If a file is consistently bad in all circumstances, it will receive a D grade, and if it's okay in two but bad in one circumstance, it will receive a B grade.

Reply:
Yes, but no score calculation is made yet, that's a good point I will lcreate an issue regarding this

Reply:
I created this issue regarding this matter #125